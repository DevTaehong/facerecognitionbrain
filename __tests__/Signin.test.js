/**
 * @jest-environment jsdom
 */
import Signin from '../src/components/Signin/Signin';
import renderer from 'react-test-renderer';

const user = {
    id: "1", 
    name: "Taehong", 
    email: "minth1123@icloud.com", 
    joined: "Today"
}

it("renders correctly", () => {
    const tree = renderer
    .create(<Signin loadUser={user} onRouteChange={"home"} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders as an anchor when no user is set', () => {
    const tree = renderer.create(<Signin />).toJSON();
    expect(tree).toMatchSnapshot();
});