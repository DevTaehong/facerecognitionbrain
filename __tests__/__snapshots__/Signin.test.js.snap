// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as an anchor when no user is set 1`] = `
<div
  className="login-form mv4 w-100 w-50-m w-25-l mw6"
>
  <div>
    <h1>
      Login
    </h1>
    <div
      className="content"
    >
      <div
        className="input-field"
      >
        <input
          id="email-address"
          name="email-address"
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
      </div>
      <div
        className="input-field"
      >
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
    </div>
    <div
      className="action"
    >
      <button
        onClick={[Function]}
      >
        Log in
      </button>
      <button
        onClick={[Function]}
      >
        Sign up
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="login-form mv4 w-100 w-50-m w-25-l mw6"
>
  <div>
    <h1>
      Login
    </h1>
    <div
      className="content"
    >
      <div
        className="input-field"
      >
        <input
          id="email-address"
          name="email-address"
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
      </div>
      <div
        className="input-field"
      >
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
    </div>
    <div
      className="action"
    >
      <button
        onClick={[Function]}
      >
        Log in
      </button>
      <button
        onClick={[Function]}
      >
        Sign up
      </button>
    </div>
  </div>
</div>
`;
